src/install-peerdeps.js

```diff
@@ -154,14 +188,18 @@ const getPackageString = ({ name, version }) => {
   if (version.indexOf(" ") >= 0) {
     // Semver ranges can have a join of comparator sets
     // e.g. '^3.0.2 || ^4.0.0' or '>=1.2.7 <1.3.0'
-    // Take the last version in the range
-    const rangeSplit = version.split(" ");
-    const versionToInstall = rangeSplit[rangeSplit.length - 1];
+    // Take the latest version in the range
+
+    const range = Range(version);
+    // Sort ranges and extract the range supporting the highest possible version
+    const rawVersionRange = range.set.sort(rangeComparator).pop();
```

Comment:
i'm not sure why this is necessary - why can't we include the entire specifier in the install command?

Reply:
Hmm we probably can, didn't think of it but it would lead to a pretty big specifier in the final package.json if we do not sort/split it (as it is we already add a fairly uncommon specifier with this method since most of them end up being carets/tildes instead of explicit ranges)

Reply:
I think that’s actually desired - people’s dep range really *should* match the peer dep 1:1

Reply:
Hmm well then if that actually works I guess there's even no need to convert the range at all, I'll try this out